e2e/connect.test.tsx

```diff
@@ -5,49 +5,60 @@
-    wallet = await component.findByTestId('wallet')
+    connectWallet = await component.findByTestId('connect-wallet')
     toolbar = await component.findByTestId('toolbar')
+    tokenSelect = (await component.findAllByTestId('token-select'))[0]
   })
 
-  it('prompts for wallet connection in the Toolbar', async () => {
-    expect(wallet.hidden).toBeTruthy()
-    expect(toolbar.textContent).toBe('Connect wallet to swap')
+  describe('with no params, using fallback JSON RPC URL', () => {
+    it('prompts for wallet connection in the Toolbar and Wallet', async () => {
+      expect(toolbar.textContent).toBe('Connecting…')
+
+      await waitFor(() => expect(toolbar.textContent).not.toBe('Connecting…'))
+      expect(connectWallet.textContent).toBe('Connect wallet to swap')
+      expect(toolbar.textContent).toBe('Connect wallet to swap')
+      expect(tokenSelect.getAttribute('disabled')).toBeFalsy()
+    })
   })
 
-  describe('with jsonRpcUrlMap', () => {
-    it('prompts for wallet connection in the Wallet', async () => {
-      act(() => component.rerender(<SwapWidget tokenList={tokens} jsonRpcUrlMap={{ 1: [hardhat.url] }} />))
-      expect(wallet.hidden).toBeTruthy()
+  describe('with integrator jsonRpcUrlMap', () => {
+    it('prompts for wallet connection in the Wallet and Toolbar', async () => {
+      component = render(<SwapWidget tokenList={tokens} jsonRpcUrlMap={{ 1: [hardhat.url] }} />)
+      expect(connectWallet.textContent).toBe('Connect wallet to swap')
       expect(toolbar.textContent).toBe('Connecting…')
 
-      act(async () => await waitFor(() => expect(toolbar.textContent).not.toBe('Connecting…')))
-      expect(wallet.hidden).toBeFalsy()
-      expect(wallet.textContent).toBe('Connect wallet to swap')
-      expect(toolbar.textContent).toBe('Enter an amount')
+      await waitFor(() => expect(toolbar.textContent).not.toBe('Connecting…'))
+      expect(connectWallet.textContent).toBe('Connect wallet to swap')
+      expect(toolbar.textContent).toBe('Connect wallet to swap')
+      expect(tokenSelect.getAttribute('disabled')).toBeFalsy()
     })
   })
 
-  describe('with provider', () => {
-    it('does not prompt for wallet connection', async () => {
-      act(() => component.rerender(<SwapWidget tokenList={tokens} provider={hardhat.provider} />))
-      expect(wallet.hidden).toBeTruthy()
+  describe('with integrator provider', () => {
+    it('does not prompt for wallet connection with hardhat', async () => {
+      component = render(<SwapWidget tokenList={tokens} provider={hardhat.provider} />)
       expect(toolbar.textContent).toBe('Connecting…')
 
-      act(async () => await waitFor(() => expect(toolbar.textContent).not.toBe('Connecting…')))
-      expect(wallet.hidden).toBeTruthy()
+      await waitFor(() => expect(toolbar.textContent).not.toBe('Connecting…'))
+      account = await component.findByTestId('account')
+      toolbar = (await component.findAllByTestId('toolbar'))[1]
```

Comment:
one thing I am very confused about is why it finds multiple elements here with testId 'toolbar' -- one that says "Connect to wallet to swap" and a second one that says "Enter an amount"